File: mobile_app/lib/models/ride.dart
class Ride {
  Ride({
    required this.id,
    required this.name,
    required this.type,
    required this.description,
    required this.image,
    required this.minAge,
    required this.capacity,
    required this.duration,
    required this.status,
  });

  final String id;
  final String name;
  final String type;
  final String description;
  final String image;
  final int minAge;
  final int capacity;
  final int duration;
  final String status;

  factory Ride.fromJson(Map<String, dynamic> json) => Ride(
        id: json['_id'] ?? '',
        name: json['ride_name'] ?? '',
        type: json['type'] ?? '',
        description: json['description'] ?? '',
        image: json['image'] ?? '',
        minAge: json['min_age'] ?? 0,
        capacity: json['capacity'] ?? 0,
        duration: json['duration'] ?? 0,
        status: json['status'] ?? '',
      );
}

File: backend/models/Ride.js
const mongoose = require("mongoose");

const rideSchema = new mongoose.Schema(
  {
    ride_name: { type: String, required: true, trim: true },
    type: { type: String, enum: ["thrill", "family", "water"], required: true },
    description: { type: String, default: "" },
    image: { type: String, default: "" },
    min_age: { type: Number, required: true, min: 0 },
    capacity: { type: Number, required: true, min: 1 },
    duration: { type: Number, required: true, min: 1 },
    status: { type: String, enum: ["Active", "Maintenance"], default: "Active" }
  },
  { timestamps: true }
);

module.exports = mongoose.model("Ride", rideSchema);
